drivers/display/display_mcux_elcdif.c

```diff
@@ -0,0 +1,286 @@
+{
+	LOG_ERR("Read not implemented");
+	return -ENOTSUP;
+}
+
+static void *mcux_elcdif_get_framebuffer(const struct device *dev)
+{
+	LOG_ERR("Direct framebuffer access not implemented");
+	return NULL;
+}
+
+static int mcux_elcdif_display_blanking_off(const struct device *dev)
+{
+	LOG_ERR("Display blanking control not implemented");
+	return -ENOTSUP;
+}
+
+static int mcux_elcdif_display_blanking_on(const struct device *dev)
+{
+	LOG_ERR("Display blanking control not implemented");
+	return -ENOTSUP;
+}
+
+static int mcux_elcdif_set_brightness(const struct device *dev,
+				      const u8_t brightness)
+{
+	LOG_WRN("Set brightness not implemented");
+	return -ENOTSUP;
+}
+
+static int mcux_elcdif_set_contrast(const struct device *dev,
+				    const u8_t contrast)
+{
+	LOG_ERR("Set contrast not implemented");
+	return -ENOTSUP;
+}
+
+static int mcux_elcdif_set_pixel_format(const struct device *dev,
+					const enum display_pixel_format
+					pixel_format)
+{
+	if (pixel_format == PIXEL_FORMAT_RGB_565) {
+		return 0;
+	}
+	LOG_ERR("Pixel format change not implemented");
+	return -ENOTSUP;
+}
+
+static int mcux_elcdif_set_orientation(const struct device *dev,
+		const enum display_orientation orientation)
+{
+	if (orientation == DISPLAY_ORIENTATION_NORMAL) {
+		return 0;
+	}
+	LOG_ERR("Changing display orientation not implemented");
+	return -ENOTSUP;
+}
+
+static void mcux_elcdif_get_capabilities(const struct device *dev,
+		struct display_capabilities *capabilities)
+{
+	const struct mcux_elcdif_config *config = dev->config->config_info;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = config->hactive;
+	capabilities->y_resolution = config->vactive;
+	capabilities->supported_pixel_formats = PIXEL_FORMAT_RGB_565;
+	capabilities->current_pixel_format = PIXEL_FORMAT_RGB_565;
+	capabilities->current_orientation = DISPLAY_ORIENTATION_NORMAL;
+}
+
+static void mcux_elcdif_isr(void *arg)
+{
+	struct device *dev = (struct device *)arg;
+	const struct mcux_elcdif_config *config = dev->config->config_info;
+	struct mcux_elcdif_data *data = dev->driver_data;
+	u32_t status;
+
+	status = ELCDIF_GetInterruptStatus(config->base);
+	ELCDIF_ClearInterruptStatus(config->base, status);
+
+	k_sem_give(&data->sem);
+}
+
+static int mcux_elcdif_init(struct device *dev)
+{
+	const struct mcux_elcdif_config *config = dev->config->config_info;
+	struct mcux_elcdif_data *data = dev->driver_data;
+	int i;
+
+	u32_t polarity_flags = kELCDIF_DataEnableActiveHigh |
+			       kELCDIF_VsyncActiveLow |
+			       kELCDIF_HsyncActiveLow |
+			       kELCDIF_DriveDataOnRisingClkEdge;
+
+	elcdif_rgb_mode_config_t rgb_mode_config = {
+		.panelWidth = config->hactive,
+		.panelHeight = config->vactive,
+		.hsw = config->hsync_len,
+		.hfp = config->hfront_porch,
+		.hbp = config->hback_porch,
+		.vsw = config->vsync_len,
+		.vfp = config->vfront_porch,
+		.vbp = config->vback_porch,
+		.polarityFlags = polarity_flags,
```

Comment:
Could polarity, pixelFomrat and databus not been taken from the config struct, hey also look like display panel specific parameters?

Reply:
Yes, done